### ISSUE TITLE
Unexpected behavior with vim margin when open NerdTree in maximum terminal.

### ISSUE THREAD

**Issue by hungpham3112** [vim/vim-win32-installer#253]
### Steps to reproduce

1.  Download my vimrc config in [here](https://github.com/hungpham3112/vide). I already have 1 installer command so you just need to paste in Powershell > 5.0.
2.  Open vim in terminal, `<space>e` to open NerdTreeToggle
3.  This unexpected behavior appears in `cmd.exe`, `powershell` terminal`, `window powershell` in maximum size, in normal size it doesn't appear. The margin of vim seems unfit with maximum terminal. I tested my config with neovim but it just happens with vim. 

cmd.exe:

https://user-images.githubusercontent.com/75968004/160745013-ca4a18a3-7c43-4333-833e-ecd9d6d66a3a.mp4

window powershell:

https://user-images.githubusercontent.com/75968004/160745017-6dd9a232-a452-485e-a51f-dcbb5242a6fe.mp4

### Expected behaviour

Vim can fit with maximum terminal. 

### Version of Vim

8.1.4643

### Environment

Window 11(64 bits).

**Comment by chrisbra** [comment]
> This unexpected behavior appears in cmd.exe, powershell terminal, window powershell` in maximum size, in normal size it doesn't appear. The margin of vim seems unfit with maximum terminal. I tested my config with neovim but it only happens with vim.

What margin? I don't see it in your videos

Also, if you want to report a bug in Vim, please:
1) provide a minimal vimrc to reproduce the bug
2) report it against vim/vim repository

**Comment by hungpham3112** [comment]
> > This unexpected behavior appears in cmd.exe, powershell terminal, window powershell` in maximum size, in normal size it doesn't appear. The margin of vim seems unfit with maximum terminal. I tested my config with neovim but it only happens with vim.
> 
> What margin? I don't see it in your videos
> 
> Also, if you want to report a bug in Vim, please:
> 
> 1. provide a minimal vimrc to reproduce the bug
> 2. report it against vim/vim repository

The green line is the "margin" I mean. Sorry, I don't know exactly what it is called.

[image](https://user-images.githubusercontent.com/75968004/160853403-aef7e797-70cc-40d3-8381-98e854af6f45.png)
[image](https://user-images.githubusercontent.com/75968004/160853510-30e23245-4a75-4645-b50e-c86c6bb5b122.png)

Minimal vimrc:
```
call plug#begin(has('nvim') ? stdpath('data') . '/plugged' : '~/vimfiles/plugged')
Plug 'https://github.com/preservim/nerdtree.git'
Plug 'https://github.com/sainnhe/edge.git', {'as': 'edge'}
Plug 'https://github.com/itchyny/lightline.vim.git'
call plug#end()

set laststatus=2
colorscheme edge
let mapleader = " "
nnoremap <leader>e :NERDTreeToggle<CR>
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
"                             Plugin: Lightline                              "
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""

let g:lightline#bufferline#unnamed = 'Unnamed'
let g:lightline#bufferline#enable_devicons = 1
let g:lightline#bufferline#buffer_filter = "LightlineBufferlineFilter"
let g:lightline = {
    \ 'colorscheme': 'edge',
    \ 'active': {
    \   'left': [ [ 'mode', 'paste' ],
    \             [ 'cocstatus', 'readonly' ],
    \             [ 'gitbranch', 'readonly', 'filename', 'modified' ] ],
    \ },
    \ 'component': {
    \   'filename': '%F',
    \ },
    \ 'component_function': {
    \   'cocstatus': 'coc#status',
    \   'filetype': 'MyFiletype',
    \   'fileformat': 'MyFileformat',
    \   'gitbranch': 'gitbranch#name',
    \ },
    \ 'tabline': {
    \   'left': [ ['buffers'] ],
    \ },
    \ 'component_expand': {
    \   'buffers': 'lightline#bufferline#buffers'
    \ },
    \ 'component_type': {
    \   'buffers': 'tabsel',
    \ },
\ }
function! MyFiletype()
    return winwidth(0) > 70 ? (strlen(&filetype) ? &filetype . ' ' . WebDevIconsGetFileTypeSymbol() : 'no ft') : ''
endfunction

function! MyFileformat()
    return winwidth(0) > 70 ? (&fileformat . ' ' . WebDevIconsGetFileFormatSymbol()) : ''
endfunction

function LightlineBufferlineFilter(buffer)
   return getbufvar(a:buffer, '&buftype') !=# 'terminal'
endfunction

if has("gui_running")
    let g:lightline.separator = { 'left': '', 'right': '' }
    let g:lightline.subseparator = { 'left': '|', 'right': '|' }
    let g:lightline.tabline_separator = g:lightline.separator
    let g:lightline.tabline_subseparator = g:lightline.subseparator
elseif has('termguicolors')
    let g:lightline.separator = { 'left': '', 'right': '' }
    let g:lightline.subseparator = { 'left': '', 'right': '' }
    let g:lightline.tabline_separator = g:lightline.separator
    let g:lightline.tabline_subseparator = g:lightline.subseparator
endif
```

**Comment by chrisbra** [comment]
isn't the margin always there, but sometimes with a default color? Don't know what is causing this, but from the video I would guess this is because your terminal cannot draw multiples of the required screen cells in either the columns or rows. That's why it adds the margin. Why it sometimes is colored green, I don't know.

**Comment by chrisbra** [comment]
can you please check if https://github.com/vim/vim/commit/5017c66bd461eab46094e076ad1aa818e246ce9d fixes it?

**Comment by hungpham3112** [comment]
> can you please check if [vim/vim@5017c66](https://github.com/vim/vim/commit/5017c66bd461eab46094e076ad1aa818e246ce9d) fixes it?

No, the issue still in version 8.2.4710. I guess this is a problem with vim redraw, it flickers a lot when resize and also in neovim it seems not affect the vertical edge.

vim:

https://user-images.githubusercontent.com/75968004/162337739-d65d5711-77a5-4390-9e21-df3c6a4d792f.mp4

nvim:

https://user-images.githubusercontent.com/75968004/162337749-69356355-ed14-4dc7-acfe-1b908061f9d6.mp4
